controllers/user.js

```diff
@@ -102,6 +104,89 @@ const updateById = (asyncHandler(async (req, res) => {
     res.send(updatedUser)
 }))
 
+// forgot password
+const forgotPassword = (asyncHandler(async (req, res) => {
+    const { email } = req.body
+    // find user has in DB user
+    const emailFounded = await userModel.findOne({ email: email })
+    // generate token hash string
+    const tokensHex = crypto.randomBytes(32).toString('hex') // 32 randomBytes will be 64 characters
+    // server router link (change if login reset password form if intergrate with client-side)
+    const serverRoutes = `${process.env.SERVER_URI}/api/auth/reset-password`
+    // create reset passwrod url for user
+    const magicLink = `${serverRoutes}?id=${emailFounded._id}&token=${tokensHex}&exp=${Date.now() + 3600000}`
```

Comment:
I think `Date.now() + 3600000`  should be stored in a var? Also, 3600000 should be in env to allow customization in the future?